Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/FindPeakAutomatic.py

```diff
@@ -0,0 +1,360 @@
+            'terminating. This is useful because sometimes good peaks can be found after '
+            'some bad ones. However setting this value too high will make the search much slower.',
+            validator=IntBoundedValidator(lower=0))
+        self.declareProperty(
+            'UsePoissonCost',
+            False,
+            doc='Use a probabilistic approach to find the cost of a fit instead of using chi2.')
+        self.declareProperty(
+            'FitToBaseline',
+            False,
+            doc='Use a probabilistic approach to find the cost of a fit instead of using chi2.')
+        self.declareProperty(
+            'EstimatePeakSigma',
+            3.0,
+            doc='A rough estimate of the standard deviation of the gaussian used to fit the peaks',
+            validator=FloatBoundedValidator(lower=0.0))
+        self.declareProperty('MinPeakSigma',
+                             0.5,
+                             doc='Minimum value for the standard deviation of a peak',
+                             validator=FloatBoundedValidator(lower=0.0))
+        self.declareProperty('MaxPeakSigma',
+                             30.0,
+                             doc='Maximum value for the standard deviation of a peak',
+                             validator=FloatBoundedValidator(lower=0.0))
+        self.declareProperty('PlotPeaks', False,
+                             'Plot the position of the peaks found by the algorithm')
+        self.declareProperty('PlotBaseline', False,
+                             'Plot the baseline as calculated by the algorithm')
+
+        # Output table
+        self.declareProperty(name='PeakPropertiesTableName',
+                             defaultValue='',
+                             doc='Name of the table containing the properties of the peaks')
+        self.declareProperty(
+            name='RefitPeakPropertiesTableName',
+            defaultValue='',
+            doc=
+            'Name of the table containing the properties of the peaks that had to be fitted twice '
+            'as the first time the error was unreasonably large')
+
+    def PyExec(self):
+        self.acceptance = self.getProperty('AcceptanceThreshold').value
+        self.smooth_window = self.getProperty('SmoothWindow').value
+        self.bad_peak_to_consider = self.getProperty('BadPeaksToConsider').value
+        self.use_poisson_cost = self.getProperty('UsePoissonCost').value
+        self.fit_to_baseline = self.getProperty('FitToBaseline').value
+        plot_peaks = self.getProperty('PlotPeaks').value
+        plot_baseline = self.getProperty('PlotBaseline').value
+        self.peak_width_estimate = self.getProperty('EstimatePeakSigma').value
+        self.min_sigma = self.getProperty('MinPeakSigma').value
+        self.max_sigma = self.getProperty('MaxPeakSigma').value
+
+        # Progress reporter that covers the whole algorithm
+        prog_reporter = Progress(self, start=0.0, end=0.1, nreports=5)
```

Comment:
why end = 0.1 and nreports =5?

Reply:
Because I split the progress bar into:
- before peak finding loop, 5 reports
- during peak finding loop, at most `num. peaks + 2` reports (but we don't know exactly a priory). On average this ends up 20-60, so I though to make it 90% of the report